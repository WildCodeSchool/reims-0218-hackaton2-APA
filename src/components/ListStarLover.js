import React from 'react';
import { ListGroup } from 'reactstrap';
import StarLover from './StarLover';


const ListStarLover = ({ data }) => <ListGroup>
       { data.map ((starlover, id) => <StarLover key={id} {...starlover} />)}
    </ListGroup>

export default ListStarLover