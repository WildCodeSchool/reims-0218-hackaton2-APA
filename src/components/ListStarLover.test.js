import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListStarLover from './ListStarLover';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ListStarLover data={[]} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  
  const props = {

     data : [
      {
     name: "MONSIEUR TARTE",
     gender: "male",
     species: "human",
     hairColor: "blond",
     eyeColor: "blue",
     skinColor: "light",
   },
   {
     name: "MONSIEUR FISCHER",
     gender: "male",
     species: "human",
     hairColor: "blond",
     eyeColor: "blue",
     skinColor: "light",
   },
   {
     name: "MONSIEUR DESCHAMPS",
     gender: "male",
     species: "human",
     hairColor: "blond",
     eyeColor: "blue",
     skinColor: "light",
   },
   {
     name: "MONSIEUR LUKE",
     gender: "male",
     species: "human",
     hairColor: "blond",
     eyeColor: "blue",
     skinColor: "light",
   }
 
  ]
  }
  
  const tree = renderer
    .create(<ListStarLover {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
