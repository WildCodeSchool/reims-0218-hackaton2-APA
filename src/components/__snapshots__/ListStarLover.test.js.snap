// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <p />
    <p>
      MONSIEUR TARTE
    </p>
    <p>
      male
    </p>
    <p>
      human
    </p>
    <p>
      blond
    </p>
    <p>
      blue
    </p>
    <p>
      light
    </p>
  </li>
  <li
    className="list-group-item"
  >
    <p />
    <p>
      MONSIEUR FISCHER
    </p>
    <p>
      male
    </p>
    <p>
      human
    </p>
    <p>
      blond
    </p>
    <p>
      blue
    </p>
    <p>
      light
    </p>
  </li>
  <li
    className="list-group-item"
  >
    <p />
    <p>
      MONSIEUR DESCHAMPS
    </p>
    <p>
      male
    </p>
    <p>
      human
    </p>
    <p>
      blond
    </p>
    <p>
      blue
    </p>
    <p>
      light
    </p>
  </li>
  <li
    className="list-group-item"
  >
    <p />
    <p>
      MONSIEUR LUKE
    </p>
    <p>
      male
    </p>
    <p>
      human
    </p>
    <p>
      blond
    </p>
    <p>
      blue
    </p>
    <p>
      light
    </p>
  </li>
</ul>
`;
